### DISCUSSION TITLE
Initial Application (Steps 1 and 2) Possible Flows

### DISCUSSION THREAD

**Discussion by penoptech** [Get-Your/Get-Your#94] [General]
@TiCampbell, I came across an issue a user could potentially find themselves in while initially going through the application. Consider the following sequence of events:

1) User creates an account in Step 1 and advances to Step 2
2) User decides to navigate "back" to Step 1.

There are a few issues that come up as a result of this flow:

1) Now that the user is back on Step 1, the CRSF token is now invalid, so a user wouldn't be able to update their information even if they tried.
2) A user might try to refresh the screen when they get no response from the application. They'll have a blank page, but if they fill out the form again, they'll get a message saying their account already exists (and the only way for them to get around this is to use a different email address so they can proceed to Step 2 again.
3) If we want to allow user's to go back from Step 2 to Step 1, I think we should prepopulate the form data with whatever they initially submitted (which we don't currently do). We'd want to put Step 1 in 'Update Mode', even though it's the initial application. This way we're not messing around with passwords (similar to how we're doing it with account updates)

I see two ways of handling this situation:

1) Don't let a user navigate back from Step 2. Easy problem to solve. I'll just conditionally hide the back button when the user is on Step 2.
2) Allow the user to go back to Step 1 from Step 2. If we go this route, we should retro the back button functionality to use hyperlinks instead of `window.history.back();`.

My preference is option 2, because I think the back button should be using hyperlinks anyway.

<img width="1457" alt="Screenshot 2023-04-09 at 3 56 51 PM" src="https://user-images.githubusercontent.com/55520784/230798530-c08b7505-89cd-44c0-a0a6-3b4ac9b6c956.png">

**Comment by TiCampbell** [comment]
@penoptech Per our Teams meeting the other day, I think we agreed that Option 2 is the way to go (and get rid of all `window.history.back()` while we're at it), but that it will be part of the redesign rather than modifying the old (current) `GetFoco` code.
